server/src/main/java/org/elasticsearch/index/search/stats/ShardSearchPhaseAPMMetrics.java

```diff
@@ -57,8 +57,8 @@ public ShardSearchPhaseAPMMetrics(MeterRegistry meterRegistry) {
     }
 
     @Override
-    public void onCanMatchPhase(long tookInNanos) {
-        recordPhaseLatency(canMatchPhaseMetric, tookInNanos);
+    public void onCanMatchPhase(ShardSearchRequest shardSearchRequest, long tookInNanos) {
+        recordPhaseLatency(canMatchPhaseMetric, tookInNanos, shardSearchRequest, null);
```

Comment:
could you add a comment about the null argument here? We may have a way to get the time range out of QueryRewriteContext. But it requires further changes. I am thinking it's not worth it, especially as for can match, I am thinking we do not need the same fine level of granularity for metrics attributes that we need for search.